src/gui/MixerView.cpp

```diff
@@ -625,8 +625,11 @@ void MixerView::updateFaders()
 	Mixer * m = Engine::mixer();
 
 	// apply master gain
-	m->mixerChannel(0)->m_peakLeft *= Engine::audioEngine()->masterGain();
-	m->mixerChannel(0)->m_peakRight *= Engine::audioEngine()->masterGain();
+	const auto masterGain = Engine::audioEngine()->masterGain();
+	auto& peakLeft = m->mixerChannel(0)->m_peakLeft;
+	auto& peakRight = m->mixerChannel(0)->m_peakRight;
+	peakLeft.store(peakLeft * masterGain);
+	peakRight.store(peakLeft * masterGain);
```

Comment:
copy/paste problem?
```suggestion
	peakRight.store(peakRight * masterGain);
```

Reply:
Yes, thank you, good eye :+1:

Reply:
Taking another look.., this is truly horrid design. The view class is changing its own model. I am not even confident in this PR anymore. As @DomClark said, we really just have to rethink the design of our systems and look at the requirements for it with fresh eyes.